copy/copy.go

```diff
@@ -894,6 +1029,44 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 		inputInfo = srcInfo
 	}
 
+	// Perform image encryption for valid mediatypes if encryptConfig provided
+	var (
+		encryptMediaType string
+		encrypted        bool
+		finalizer        ocicrypt.EncryptLayerFinalizer
+	)
+	if toEncrypt {
+		encryptMediaType, err = manifest.GetEncryptedMediaType(srcInfo.MediaType)
+		if err != nil {
+			return types.BlobInfo{}, errors.Wrapf(err, "Error encrypting layer %s", srcInfo.Digest)
+		}
+
+		if encryptMediaType != "" && c.encryptConfig != nil {
+			var annotations map[string]string
+			if !decrypted {
+				annotations = srcInfo.Annotations
+			}
+			desc := ocispec.Descriptor{
+				MediaType:   srcInfo.MediaType,
+				Digest:      srcInfo.Digest,
+				Size:        srcInfo.Size,
+				Annotations: annotations,
+			}
+
+			s, fin, err := ocicrypt.EncryptLayer(c.encryptConfig, destStream, desc)
+			if err != nil {
+				return types.BlobInfo{}, errors.Wrapf(err, "Image encryption failed for the digest %+v", srcInfo.Digest)
+			}
+
+			destStream = s
+			finalizer = fin
+			inputInfo.Digest = ""
+			inputInfo.Size = -1
+			inputInfo.MediaType = encryptMediaType
```

Comment:
This should not be something `copy.Image` does; let `UpdatedImage` handle that, like compression.  Instead of `GetEncryptedMediaType`, have `types.Image.CanEncryptLayer`==`genericManifest.CanEncryptLayer`, and leave the MIME types to the manifest implementations.

Reply:
… “This” being editing the MIME type value.

Reply:
Gotcha - i think this is an artifact from before we had added fields to BlobInfo, so this can be removed.

Reply:
Removed